spec/with_locking_spec.rb

```diff
@@ -48,10 +49,40 @@
 
     it "checks if the file exists" do
       File.should_receive(:exists?).with(path)
+      File.should_receive(:exists?).with(path)
```

Comment:
Oh, read the docs on [Receive Counts](https://relishapp.com/rspec/rspec-mocks/v/3-5/docs/setting-constraints/receive-counts). You want to call `twice` to indicate this method should be called multiple times.

I feel like calling `#exists?` is an implementation detail. Whether the implementation calls `#exists?` once or twice or 100 times shouldn't matter. I feel like `#exists?` should be stubbed, not mocked.